org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/xtext/index/persistence/GitService.xtend

```diff
@@ -29,12 +38,17 @@ class GitService {
 
 	@Accessors(PUBLIC_GETTER)
 	File projectFolder
-
+	
+	String privateKeyLocation
+	String knownHostsLocation
+	
 	/** 
 	 * initialize this git service. either open the existing git and pull, or clone the remote repo
 	 */
-	def void init(String localRepoFileRoot, String remoteRepoUrl) {
+	def void init(String localRepoFileRoot, String remoteRepoUrl, String privateKeyLocation, String knownHostsLocation) {
```

Comment:
I suppose this is to keep `GitService` independent from Dropwizard-specifics? (Otherwise, we could just accept an instance of the corresponding `…Config` class.)

But consider overloading the method:

    def void init(String localRepoFileRoot, String remoteRepoUrl) {
        init(localRepoFileRoot, remoteRepoUrl, null, null)
    }